doc/user/content/transform-data/patterns/percentiles.md

```diff
@@ -10,134 +10,306 @@ menu:
+domains with high cardinality, because of the cross join to calcuate
+distributions, the memory usage can become significant. As such, using
+histograms to compute percentiles are better suited for domains with _low_
+cardinality.
+
+To use histograms to compute percentiles:
+
+- First, create a histogram view that tracks the unique values and their
+  associated count/frequency.
+
+- Then, using a cross join on the histogram view, create a distribution view
+  that calculates the distribution of the values. The distribution is calculated
+  by dividing the cumulative frequency (the count of all values up to and
+  including that bucket) by the total counts for all bucket values.
+
+  {{< note >}}
+
+  Because of the cross join (which grows quadratic in the number of unique
+  values), the required memory may grow large for domain with high cardinality.
+
+  {{</ note >}}
+
+### Example
+
+1. Create a table `input`:
+
+   ```mzsql
+   CREATE TABLE input (value BIGINT);
+   ```
+
+2. Insert into the `input` table values `1` to `10`.
+
+   ```mzsql
+   INSERT INTO input
+   SELECT n FROM generate_series(1,10) AS n;
+   ```
+
+1. Create a `histogram` view to track unique values from the
+   `input` table and their frequency/counts:
+
+   ```mzsql
+   CREATE VIEW histogram AS
+   SELECT
+     value AS bucket,
+     count(*) AS frequency
+   FROM input
+   GROUP BY value;
+   ```
+
+1. Create a view `distribution` to calculate the cumulative frequency/count and
+   the cumulative distribution. The view definition uses a cross join on the
+   `histogram` view. The distribution is calculated by dividing the cumulative
+   frequency (the count of all values up to and including that bucket) by the
+   total counts for all bucket values.
+
+   ```mzsql
+   CREATE VIEW distribution AS
+   SELECT
+     h.bucket,
+     h.frequency,
+     sum(g.frequency) AS cumulative_frequency,
+     sum(g.frequency) / (SELECT sum(frequency) FROM histogram) AS  cumulative_distribution
+   FROM histogram g, histogram h
+   WHERE g.bucket <= h.bucket
+   GROUP BY h.bucket, h.frequency
+   ORDER BY cumulative_distribution;
+   ```
+
+   {{< note >}}
+
+   This view uses a self-join that scales quadratically with the number of
+   unique bucket values. If the data has high cardinality, memory usage may grow
+   significantly.
+
+   {{</ note >}}
+
+1. You can then query `distribution` by the `cumulative_distribution` field to
+   return specific percentiles. For example, the following query returns the
+   90-th percentile.
+
+   ```mzsql
+   SELECT bucket AS percentile90
+   FROM distribution
+   WHERE cumulative_distribution >= 0.9
+   ORDER BY cumulative_distribution
+   LIMIT 1;
+   ```
+
+1. To keep the `distribution` view up to date as new data comes into `input`,
+   you can create an index on the view. To optimize for percentile queries,
+   index on the `cumulative_distribution` column to support [point lookup on
```

Comment:
I feel like this index may not help much, and may just confuse the reader. The above query would work well, but only if indexes were ordered rather than hash indexes. I think creating such an index would not improve any queries other than ones where you know the numeric `cumulative_distribution` value you are looking for, which seems rare.

Reply:
removed in patch 75f32afbf3839b0dc7b5f84321847c6c88161bff